applications/ThermalDEMApplication/thermal_dem_application_variables.cpp

```diff
@@ -0,0 +1,80 @@
+  KRATOS_CREATE_VARIABLE(std::string,                          VORONOI_METHOD_NAME)
+  KRATOS_CREATE_VARIABLE(std::string,                          POROSITY_METHOD_NAME)
+  KRATOS_CREATE_VARIABLE(bool,                                 MOTION_OPTION)
+  KRATOS_CREATE_VARIABLE(bool,                                 DIRECT_CONDUCTION_OPTION)
+  KRATOS_CREATE_VARIABLE(bool,                                 INDIRECT_CONDUCTION_OPTION)
+  KRATOS_CREATE_VARIABLE(bool,                                 CONVECTION_OPTION)
+  KRATOS_CREATE_VARIABLE(bool,                                 RADIATION_OPTION)
+  KRATOS_CREATE_VARIABLE(bool,                                 FRICTION_HEAT_OPTION)
+  KRATOS_CREATE_VARIABLE(bool,                                 REAL_CONTACT_OPTION)
+  KRATOS_CREATE_VARIABLE(bool,                                 TEMPERATURE_DEPENDENT_RADIUS_OPTION)
+  KRATOS_CREATE_VARIABLE(bool,                                 FIXED_TEMPERATURE)
+  KRATOS_CREATE_VARIABLE(bool,                                 ADIABATIC)
+  KRATOS_CREATE_VARIABLE(int,                                  THERMAL_FREQUENCY)
+  //KRATOS_CREATE_VARIABLE(double,                             HEATFLUX)             // defined in DEMApp
+  //KRATOS_CREATE_VARIABLE(double,                             THERMAL_CONDUCTIVITY) // defined in DEMApp
+  KRATOS_CREATE_VARIABLE(double,                               REAL_YOUNG_MODULUS_RATIO)
+  KRATOS_CREATE_VARIABLE(double,                               HEATSOURCE)
+  KRATOS_CREATE_VARIABLE(double,                               MIN_CONDUCTION_DISTANCE)
+  KRATOS_CREATE_VARIABLE(double,                               MAX_CONDUCTION_DISTANCE)
+  KRATOS_CREATE_VARIABLE(double,                               ISOTHERMAL_CORE_RADIUS)
+  KRATOS_CREATE_VARIABLE(double,                               MAX_RADIATION_DISTANCE)
+  KRATOS_CREATE_VARIABLE(double,                               FRICTION_HEAT_CONVERSION)
+  KRATOS_CREATE_VARIABLE(double,                               AVERAGE_POROSITY)
+  KRATOS_CREATE_VARIABLE(double,                               ALPHA_SHAPE_PARAMETER)
+  KRATOS_CREATE_VARIABLE(double,                               INTEGRAL_TOLERANCE)
+  KRATOS_CREATE_VARIABLE(double,                               FLUID_LAYER_THICKNESS)
+  KRATOS_CREATE_VARIABLE(double,                               FLUID_DENSITY)
+  KRATOS_CREATE_VARIABLE(double,                               FLUID_VISCOSITY)
+  KRATOS_CREATE_VARIABLE(double,                               FLUID_THERMAL_CONDUCTIVITY)
+  KRATOS_CREATE_VARIABLE(double,                               FLUID_HEAT_CAPACITY)
+  KRATOS_CREATE_VARIABLE(double,                               FLUID_TEMPERATURE)
+  KRATOS_CREATE_VARIABLE(Vector,                               FLUID_VELOCITY)
+
+  // Flags (starting at 1000 to avoid conflicts with DEMFlags)
+  KRATOS_CREATE_LOCAL_FLAG(DEMThermalFlags, IS_ADIABATIC,                     1000);
```

Comment:
1000? I think there is only room for 64 bits. 32 are already used by the Kratos Core and the other 32 are partially used by the DEM.

Reply:
It is weird that you did not have problems changing bit number 1000 of a variable which has 64 bits.

Reply:
I thought these flags were integers that we could give any value, but I noticed I get some problems if I repeat the same value from the DEM App, so I changed them for a high value that the DEM App would not reach. Really strange that it worked then.
For now, I am starting from 19, but soon I think will stop using these flags, it is less risky.

Reply:
I removed all the flags, but the IS_ADIABATIC because this one can be set also for the walls, and it would require a thermal wall sub-class just for adding a variable for this flag.